### DISCUSSION TITLE
Using AWS Credentials Directly in Amazon Bedrock Custom Tool

### DISCUSSION THREAD

**Discussion by TalentumSebasG** [langflow-ai/langflow#1706] [Q&A]
Hello everyone,

I'm currently working on integrating AWS credentials directly into Amazon Bedrock, a custom tool we're utilizing in our project. I've been facing an issue where attempting to use AWS credentials directly in the Amazon Bedrock tool without creating an environment variable leads to an error. Here's the specific context:

File Structure:
[https://github.com/langflow-ai/langflow/blob/dev/src/backend/base/langflow/components/model_specs/AmazonBedrockSpecs.py](url)
This file contains the specifications for our Amazon Bedrock model.

[https://github.com/langflow-ai/langflow/blob/dev/src/backend/base/langflow/components/models/AmazonBedrockModel.py](url)
This file is where I'm encountering the error. The error message reads:

```
Invalid type conversion. Please check your code and try again.
Traceback (most recent call last):
  File "/home/power/.pyenv/versions/3.10.12/lib/python3.10/site-packages/langflow/interface/custom/utils.py", line 298, in build_custom_component_template
  ...
  ImportError: cannot import name 'Text' from 'langflow.field_typing'
Issue:
```

The main issue I'm facing is integrating AWS credentials directly into Amazon Bedrock without relying on environment variables.

I would greatly appreciate any insights, suggestions, or guidance from the community on how to resolve this issue and effectively integrate AWS credentials into Amazon Bedrock. Additionally, if anyone has encountered a similar challenge or has experience with custom tools and AWS integration, your expertise would be invaluable.

Thank you in advance for your assistance!

**Comment by Cristhianzl** [comment]
Hi! 👋

We are using other channels as our official means of communication with users. We apologize for the delayed response.
For faster assistance, please direct your questions to our Discord channel or open a GitHub issue, as we are much more active on these platforms.

[Langflow Discord Server](https://discord.com/invite/EqksyE2EX9)

Thank you for your understanding.

Best regards,
The Langflow Team
